rs/messaging/src/routing/stream_handler.rs

```diff
@@ -793,7 +773,7 @@ impl StreamHandlerImpl {
         msg_type: &str,
         state: &mut ReplicatedState,
         available_guaranteed_response_memory: &mut i64,
-    ) -> Result<(), (StateError, RequestOrResponse)> {
+    ) -> Option<(RejectReason, RequestOrResponse)> {
```

Comment:
Returning `Result` here was a bit awkward since it would return `Ok` for cases where it failed to induct but we didn't produce a reject response, i.e. it didn't mean 'we successfully inducted the message'. Essentially I had made it like this because then it was using the same return type `ReplicatedState::push_input()` does and we needed both the message and the `StateError` to generate the reject response.
But now it returns a reject reason instead, well basically a reject signal. So the function now may or may not return a reject signal, which says nothing about whether we failed or logged an error.

The idea was initially to have the only change be in this place for this PR by replacing generating a reject response with pushing a reject signal, but now it seems generating the signal inside `induct_message_impl()` makes more sense.

Reply:
Had to change it again because of supporting the old and the new way simultaneously. The return value is still optional though and I added a doc comment explaining this choice.